source/MRMesh/MRCone3.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include "MRLine.h"
+
+namespace MR
+{
+
+// Base class for cone parameterization
+
+template <typename T>
+class Cone3
+{
+public:
+    Cone3()
+    {}
+
+    // inAxis -- apex position and main axis direction.  For any cone point dot( point , direction ) >=0
+    // inAngle -- cone angle, main axis vs side
+    // inHeight -- cone inHeight
+    // main axis direction could be non normalized.
+    Cone3( const Line3<T>& inAxis, T inAngle, T inHeight )
+        :
+        axis( inAxis ),
+        angle( inAngle ),
+        height( inHeight )
+    {}
+
+    MR::Vector3<T>& center( void )
+    {
+        return axis.p;
+    }
+
+    const MR::Vector3<T>& center( void ) const 
+    {
+        return axis.p;
+    }
```

Comment:
please add coments to specify that we count apex as center (its not clear if it is apex or base, or middle point on height)

Reply:
ok